com.playeveryware.eos/Documentation~/unit-testing.md

```diff
@@ -0,0 +1,148 @@
+2. **Act**: Execute the method or function you are testing.
+3. **Assert**: Verify that the results are what you expect.
+
+```csharp
+[TestFixture]
+public class ExampleTest
+{
+    [Test]
+    public void Addition_ShouldReturnCorrectSum()
+    {
+        // Arrange
+        int a = 5;
+        int b = 3;
+        
+        // Act
+        int result = a + b;
+        
+        // Assert
+        Assert.AreEqual(8, result);
+    }
+}
+```
+
+### Coverage Requirements
+
+Any **new functionality** added to the project should come with complete unit test coverage. The goal is to ensure that every feature introduced is tested thoroughly. 
+
+When **refactoring** existing code, make sure that the unit tests are updated accordingly. Refactors should never lead to a reduction in test coverage.
+
+### Handling Dependencies
+
+While there is no current implementation for **mocking the EOS SDK**, this functionality is in development. Once mocking support is available, you will be able to write unit tests that mock dependencies on the EOS SDK.
+
+For now, focus on isolating business logic and core plugin functionality in your tests.
+
+## Moving Forward: Code Coverage Goals
+
+As we transition fully to NUnit and increase our test coverage, the goal is to reach and maintain high code coverage across the entire project. To facilitate this, developers should:
+- Write tests for all new code.
+- Update tests when refactoring or modifying existing code.
+- Ensure tests are well-structured, easy to read, and maintainable.
+
+## Best Practices for Unit Tests
+
+- **Small, isolated tests**: Each test should focus on a single function or method.
+- **Meaningful names**: Test methods should clearly describe the scenario being tested.
+- **Consistency**: Follow a consistent structure for arranging, acting, and asserting in your tests.
+- **Avoid testing too much at once**: Unit tests should focus on small, testable components, not entire systems.
+- **Repeatable and fast**: Tests should be able to run quickly and consistently without dependencies on external services or the environment.
+
+## Naming Conventions for Unit Tests
+
+To maintain clarity and consistency, unit test names should follow these conventions:
+
+1. **Use Descriptive Names**: Test names should clearly describe the scenario being tested and the expected outcome. Follow the pattern:
+   `MethodName_Scenario_ExpectedOutcome`
+   
+   - Example: `CalculateTotal_WithValidInputs_ReturnsCorrectSum()`
+
+2. **PascalCase**: Use PascalCase for test method names, capitalizing each word.
+   - Example: `Login_WithInvalidPassword_ThrowsException()`
+
+3. **Group Tests by Class or Method**: Create test classes named after the class being tested, with `Tests` appended.
+   - Example: `OrderProcessorTests`
+
+4. **Arrange-Act-Assert**: Use comments or whitespace to clearly separate the three stages of each test.
+   - Example:
+   ```csharp
```

Comment:
I'm surprised this works; you didn't define a end triple backtick thing, yet the markdown knows to end it properly after the double line break... But for my sanity, can you put an ending triple backtick?